File: backend/src/main/java/com/example/backendapplication/model/Task.java
package com.example.backendapplication.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.UUID;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tasks")
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String name;

    private String details;

    private boolean completed;
}

File: frontend/src/types/Task.ts
import { SubTask } from "./SubTask";

export type Task = {
  id: string;
  details?: string;
  completed: boolean;
  createdDate?: string;
  updatedDate?: string;
  subTasks?: SubTask[];
} & CreateTask;

export type CreateTask = {
  name: string;
};
